connectors-e2e-test/connectors-e2e-test-servicenow/src/test/java/io/camunda/connector/e2e/ServiceNowTests.java

```diff
@@ -0,0 +1,123 @@
+package io.camunda.connector.e2e;
+
+import io.camunda.client.CamundaClient;
+import io.camunda.client.api.response.DeploymentEvent;
+import io.camunda.client.api.response.ProcessInstanceResult;
+import io.camunda.connector.e2e.app.TestConnectorRuntimeApplication;
+import io.camunda.zeebe.model.bpmn.Bpmn;
+import io.camunda.zeebe.model.bpmn.BpmnModelInstance;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.core.env.Environment;
+
+import java.io.File;
+import java.util.function.Function;
+
+@SpringBootTest(
+    classes = {TestConnectorRuntimeApplication.class},
+    properties = {
+        "spring.main.allow-bean-definition-overriding=true"
+    },
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT
+)
+public class ServiceNowTests {
+  @Autowired
+  private CamundaClient client;
+
+  @Autowired
+  Environment env;
+
+  @TempDir
+  File tempDir;
+
+  String elementTemplatePath =
+      "../../connectors/servicenow/element-templates/servicenow-connector.json";
+
+  /**
+   * Mutable pipeline context for passing function compositions
+   */
+  private static class PipelineContext {
+    final String processId;
+    final String serviceTaskName;
+    final File filledTemplate;
+    BpmnModelInstance model;
+    BpmnModelInstance updatedModel;
+    DeploymentEvent deploymentEvent;
+    ProcessInstanceResult processInstanceResult;
+
+    PipelineContext(String processId, String serviceTaskName, File filledTemplate) {
+      this.processId = processId;
+      this.serviceTaskName = serviceTaskName;
+      this.filledTemplate = filledTemplate;
+    }
+  }
```

Comment:
Why not having a simple record class with methods like `getModel`, `updateModel`,... seems like an abstraction that (from a reader perspective) adds more complexity than it solves.

Reply:
I just like the .andThen() chain this enables :) But I get your point - will refactor

Reply:
@vobu `then()` you would enjoy https://effect.website/